pkg/data/values.go

```diff
@@ -18,18 +27,43 @@ func GetValueN(data map[string]interface{}, keys ...string) interface{} {
 	return val
 }
 
-func GetValue(data map[string]interface{}, keys ...string) (interface{}, bool) {
+// GetValue gets a value from the provided data field, which must be a map[string]interface or a []interface. Keys can
+// be the string keys (in order of precedence) or can be the index (as a string) of the item to be retrieved. Returns
+// the retrieved value (if any) and a bool indicating if they value was found.
+func GetValue(data interface{}, keys ...string) (interface{}, bool) {
 	for i, key := range keys {
 		if i == len(keys)-1 {
-			val, ok := data[key]
-			return val, ok
+			if dataMap, ok := data.(map[string]interface{}); ok {
+				val, ok := dataMap[key]
+				return val, ok
+			}
+			if dataSlice, ok := data.([]interface{}); ok {
+				return itemByIndex(dataSlice, key)
+			}
+		}
+		if dataMap, ok := data.(map[string]interface{}); ok {
+			data, _ = dataMap[key]
+		} else if dataSlice, ok := data.([]interface{}); ok {
+			data, _ = itemByIndex(dataSlice, key)
 		}
-		data, _ = data[key].(map[string]interface{})
 	}
 
 	return nil, false
 }
 
+func itemByIndex(dataSlice []interface{}, key string) (interface{}, bool) {
+	keyInt, err := strconv.Atoi(key)
+	if err != nil {
+		return nil, false
+	}
+	if keyInt >= len(dataSlice) || keyInt < 0 {
+		return nil, false
+	}
+	return dataSlice[keyInt], true
+}
+
+// PutValue puts val into data (in place) at the index specified by keys (specified in order of precedence). If there
```

Comment:
```suggestion
// PutValue updates the value of a given map at the index specified by keys that denote the path to the value in the nested structure of the map.
// If there is no current entry at a key, a new map is created for that value.
```

Reply:
I've made some adjustments to the doc comment. LMK if you want any further changes.